WooCommerce/Classes/ViewRelated/CardPresentPayments/CardReaderConnectionController.swift

```diff
@@ -418,7 +466,15 @@ private extension CardReaderConnectionController {
                 let properties = reader.batteryLevel
                     .map { ["battery_level": $0] }
                 ServiceLocator.analytics.track(.cardReaderConnectionSuccess, withProperties: properties)
-                self.returnSuccess(connected: true)
+                // If we were installing a software update, introduce a small delay so the user can
+                // actually see a success message showing the installation was complete
+                if case .updating(progress: 1) = self.state {
```

Comment:
It did feel a bit uncomfortable, but we are setting this value to `1` manually [here](https://github.com/woocommerce/woocommerce-ios/pull/4974/files/4129293b7734170dd21a41b748beb3653af84f38#diff-34012b275288124a9a1c606592b1c55684086c11bf413ab96e2f067fee680f67R454), when we receive a `.completed` event.

I think every time that value is 1 will be because we set a literal `1` in the code. And every time we check, we check against a literal `1`, there are no extra operations there. I'm not an expert in dealing with floating point issues, but it feels safe to me

Reply:
True... for the `1` we set we are probably safe, since we can very likely able to depend on the Swift compiler to encode the `1` the same way twice, and then isEqual should be comparing two identical IEEE-754 floats 🤞 

(Ref https://developer.apple.com/documentation/swift/float/1845934-isequal and https://ieeexplore.ieee.org/document/8766229)